src/test/ui/associated-types/associated-types-binding-to-type-defined-in-supertrait.stderr

```diff
@@ -1,20 +1,26 @@
 error[E0271]: type mismatch resolving `<ModelT as Vehicle>::Color == Blue`
   --> $DIR/associated-types-binding-to-type-defined-in-supertrait.rs:31:10
    |
+LL | impl Vehicle for ModelT { type Color = Black; }
+   |                                        ----- expected this to be `Blue`
```

Comment:
Personally, I find error messages hard to follow when they start with 'context', as opposed to the location where the error occurs. I think a better order would be:

`blue_var(ModelT) // type mismatch resolving`
`fn blue_car // required by this bound`
`impl Vehicle for ModelT` // expected this to be `lue`

Reply:
We could change the order by leveraging a new `note` instead of going straight to `span_label`s...

Reply:
Is this closer to what you'd desire?
<img width="725" alt="linked error using notes instead of labels" src="https://user-images.githubusercontent.com/1606434/126399337-b667cc45-5965-4aa0-9845-3780715b0adc.png">

Reply:
Added in the last commit, but I feel like we should separate this in a different PR, to make it easier for the reviewer.